crates/primitives/src/block.rs

```diff
@@ -204,72 +152,83 @@ impl<'a> arbitrary::Arbitrary<'a> for Block {
 
 /// Sealed block with senders recovered from transactions.
 #[derive(Debug, Clone, PartialEq, Eq, Default, Deref, DerefMut)]
-pub struct BlockWithSenders {
+pub struct BlockWithSenders<B: reth_primitives_traits::Block = Block> {
```

Comment:
```suggestion
pub struct BlockWithSenders<B = Block> {
```
it's always better to not constraint the generic in the type definition with a trait bound that implements a bunch of behaviour (ie trait methods), makes the code more flexible in testing

Reply:
updated in f7c301d88